addons/product/models/product.py

```diff
@@ -569,6 +569,19 @@ def view_header_get(self, view_id, view_type):
             )
         return super().view_header_get(view_id, view_type)
 
+    def action_open_label_layout(self):
+        view = self.env.ref('product.product_label_layout_form')
+        return {
+            'name': _('Choose Labels Layout'),
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'res_model': 'product.label.layout',
+            'views': [(view.id, 'form')],
+            'view_id': view.id,
+            'target': 'new',
+            'context': {'default_product_id': self.id},
+        }
```

Comment:
You can call the template's method and override the context
```suggestion
        self.ensure_one()
        action = self.product_tmpl_id.action_open_label_layout()
        action['context'] = {'default_product_id': self.id}
        return action
```